### PR TITLE
repository maintance

### PR DESCRIPTION
**Pull request by MrFoxPro**
I did some work to refactor and update this plugin
- Rewrote plugin to typescript (Package can be built with `pnpm build`)
- Removed explicit `?url` handling, as it supports Vite by default
- Removed `[name]` pattern, as Vite supports globs by default: https://vitejs.dev/guide/features.html#glob-import
- Removed `app-js` example, as it's very similiar to `app-ts`
- Rewrote examples, combined shared config and assets
- Changed Readme accordingly to this changes
- Added prettierrc
- Changed package.json to include compiled files and typedef. **PLEASE CHECK IT BEFORE PUBLISHING**

I've checked on my [project](https://github.com/mrfoxPro/bloki) and examples, please check it on your setup and your projects.

### CODE REVIEW

**Comment by MrFoxPro** [comment]
Closes https://github.com/jfgodoy/vite-plugin-solid-svg/issues/21

**Comment by MrFoxPro** [comment]
@jfgodoy

**Comment by jfgodoy** [comment]
wow @MrFoxPro you rock!! I really appreciate the time you have taken to improve this plugin

**Comment by MrFoxPro** [comment]
I'm thinking of changing `?component` to `?solid`, because I need to transform svg in both solid and astro, so this approach can help distinguish between renderers

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  plugins: [
+    del({ targets: 'dist/*' }),
+    babel({
+      extensions: ['.ts'],
+      babelHelpers: 'bundled',
+      presets: ['@babel/preset-typescript'],
+      exclude: /node_modules\//,
+    }),
+    copy({
+      targets: [{ src: './package/types.d.ts', dest: './dist' }],
```
Comment (jfgodoy) [inline rollup.config.js:32]: the `types.d.ts` file is in the root

File: README.md [hunk 2]
```diff
@@ -53,8 +64,11 @@ If you need to configure `svgo`, you can also create a config file `svgo.config.
 ### Usage
 
 Import as a Solid.js component:
-```
-import MyIcon from './svgs/my-icon.svg';
+
+```tsx
+import MyIcon from './my-icon.svg';
+// or './my-icon.svg' if `defaultAsComponent` is `false`
```
Comment (jfgodoy) [inline README.md:70]:
```suggestion
// or './my-icon.svg?component' if `defaultAsComponent` is `false`
```

**Comment by jfgodoy** [comment]
I'm thinking about removing the default mode, as it has issues with typescript inference. What do you think?

File: rollup.config.js (hunk 1 above)
Comment (MrFoxPro) [reply rollup.config.js]: Right. Removed this command and plugin too.

File: README.md (hunk 2 above)
Comment (MrFoxPro) [reply README.md:70]: I changed `component` to `component-solid`. In my personal case, I need in one project load svg as Astro components in one place, and as Solid components and another. So Vite need to distinguish which plugin to use.

**Comment by MrFoxPro** [comment]
> I'm thinking about removing the default mode, as it has issues with typescript inference. What do you think?

There is a slight inconsistency, I do not think it is critical.

**Comment by MrFoxPro** [comment]
@jfgodoy please publish to npm when you'll have time

**Comment by jfgodoy** [comment]
@MrFoxPro, thank you very much, you did a great job!!
I just published the package to npm as 0.6.0

**Pull request merged**